render/render.js

```diff
@@ -489,18 +489,18 @@ module.exports = function($window) {
 	}
 
 	//lifecycle
-	function initLifecycle(source, vnode, hooks) {
-		if (typeof source.oninit === "function") source.oninit.call(vnode.state, vnode)
-		if (typeof source.oncreate === "function") hooks.push(source.oncreate.bind(vnode.state, vnode))
+	function initLifecycle(vnode, source, hooks) {
```

Comment:
Leo advises [against](https://github.com/lhorie/mithril.js/blob/e334b5026fbed327542b595b5ab0ae09f6d8d259/CONTRIBUTING.md#what-should-i-know-in-advance-when-attempting-a-performance-related-contribution) caching property accesses in temp variables although it may be different here because it is a `dynamic[access]`. It would need a benchmark. I have benchmarked a previous incarnation of this PR but lookups were still static and it had no effect, whereas this change may be detrimental. 

@lhorie beside the test suite and dbmon, what do you use to watch for perf regressions?

Agreed regarding `prop` vs `source`. Updating.